### DISCUSSION TITLE
Allow resetting of metadata plots

### DISCUSSION THREAD

**Discussion by clayheaton** [dagster-io/dagster#22526] [Ideas]
In handling a last-minute request, I implemented a flow that, among other things, outputs an integer as metadata, for tracking load of users into a system. Following initial deployment and running, I learned more about timing, etc. Unfortunately, the metadata plot was looking janky because of the initial deployment schedule. Not knowing how to remove the data recorded in the first runnings, I updated the flow to create a new variable, hoping the old plot would disappear. 

![Screenshot 2024-06-13 at 9 54 32 AM](https://github.com/dagster-io/dagster/assets/205364/2b5f957b-8562-4d75-ada8-4847eebda152)

Now that things have stabilized, I'd like to be able to reset the top plot to get rid of the noise and to remove the bottom plot. I do not see where this is possible through the UI or in code. Does it require a manual database removal of rows?

This is a suggestion for a feature that allows for the resetting of metadata plots on assets.

**Comment by garethbrickman** [comment]
Yes you would remove the run records from the database. There's an explainer how to do that in https://github.com/dagster-io/dagster/discussions/12047

**Comment by ehb-mtk** [comment]
Suppose that the run records are desired but the metadata is not (in our case: it is a date-partitioned asset; there was a bug in the code on July 11th causing us to collect irrelevant metadata, and we don't want to see a plot with one bullet point on July 11th indefinitely in the UI alongside our other plots). Is there a direct API to delete just the metadata associated with a particular run? Alternatively, how dangerous would it be to hop onto the DB and drop a couple rows from a table? (I posted the same thing in #12047 before realizing this was the more logical spot for it.)

**Reply by garethbrickman** [reply]
There is no direct API to delete just the metadata of a run. It would be dangerous to drop rows directly from the database table without thorough testing.

**Reply by kevin-longe-unmind** [reply]
Would it be possible for there to be some option where we define which metadata we want visualised as plots in the asset definition? Being in a position where accidentally naming something wrong causes a plot to be stuck there forever is quite frustrating (and depending on how bad the mistake, can make the plots area unusable). Deleting run records from the database seems too heavy handed as an alternative
